### ISSUE TITLE
`Sep` widget inconsistent rendering on `0.20` release

### ISSUE THREAD

**Issue by jamestrew** [qtile/qtile#3272]
### Description
The `Sep` widget fails to render correctly when used via an assigned variable.
eg:
```python
basic_sep = Sep(foreground="#1e222a", linewidth=4)
line_sep = Sep(foreground="#abb2bf", linewidth=1, padding=10)
```
I've been experiencing this since the `0.20` release, and rolling back to `0.19` resolves this issue.

### Steps to reproduce
1. Be on `0.20` release of qtile
2. Use [this](https://gist.github.com/jamestrew/06dbc832a259edf55ca5121fbf5e9f74#file-config-py) as part of your Qtile config.py (replacing your `widget_defaults` and `screens`)
3. Restart Qtile

### Expected Behavior
Click into and zoom in for better look.
![image](https://user-images.githubusercontent.com/66286082/151672628-687997e1-7088-4006-821c-539bfca25813.png)
The `Sep` widget renders with the correct `foreground`, `linewidth` and `padding`.

### Actual Behavior
Click into and zoom in for better look.
![image](https://user-images.githubusercontent.com/66286082/151672765-2bd32b10-f166-430f-8e74-736f518c976b.png)
Notice on the left hand side, the `Sep` widgets are rendering transparently. Moving to the right, they seem to fail to render at all, causing all sorts of clashing between the other widgets. The two `Sep` widgets (`line_sep` left of the `Clock` and `basic_sep` after the `Systray`)  finally renders as indented.

Replacing each of the `line_sep` and `basic_sep` with `Sep(foreground="#abb2bf", linewidth=1, padding=10)` and `Sep(foreground="#1e222a", linewidth=4)` respectively resolves this issue. [Try here](https://gist.github.com/jamestrew/092c78fb4d10c757b7839f05dc2e916e).

Interestingly, I also tried both `0.19` and `0.20` with & without a second monitor and on my second monitor, the `Sep` widgets render correctly in all scenarios.
I suspect this issue has something to do with the render orders.

### System Info
System: `Linux 5.16.3-arch1-1`

@elParaguayo I noticed many of your PRs on the widgets where incorporated into the latest release, maybe you have an idea of what may be happening here?

**Comment by elParaguayo** [comment]
When you use a named instance multiple times, it becomes `Mirror` object so something's obviously gone wrong in the logic there but I haven't touched that class for quite a while.

Are you able to do a `git bisect` to isolate the offending commit?

**Comment by jamestrew** [comment]
I previously had Qtile installed through `pacman` and now I've cloned the repo and ran `pip install .` in the repo directory.

```
~/Documents/projects/qtile #v0.19.0 ❯ pip install .                                                                                                                                                                             12:59:13
Defaulting to user installation because normal site-packages is not writeable
Processing /home/jt/Documents/projects/qtile
Requirement already satisfied: cffi>=1.1.0 in /usr/lib/python3.10/site-packages (from qtile==0.19.0) (1.15.0)
Requirement already satisfied: cairocffi[xcb]>=0.9.0 in /home/jt/.local/lib/python3.10/site-packages (from qtile==0.19.0) (1.3.0)
Requirement already satisfied: xcffib>=0.10.1 in /home/jt/.local/lib/python3.10/site-packages (from qtile==0.19.0) (0.11.1)
Requirement already satisfied: pycparser in /usr/lib/python3.10/site-packages (from cffi>=1.1.0->qtile==0.19.0) (2.21)
Requirement already satisfied: six in /usr/lib/python3.10/site-packages (from xcffib>=0.10.1->qtile==0.19.0) (1.16.0)
Building wheels for collected packages: qtile
  Building wheel for qtile (setup.py) ... done
  Created wheel for qtile: filename=qtile-0.19.0-cp310-cp310-linux_x86_64.whl size=512825 sha256=b49d5de379d81eeeeee4a45c4021675663f427f6367e70a0a8e9b6b6b59c3478
  Stored in directory: /tmp/pip-ephem-wheel-cache-wfbsxoup/wheels/d3/10/95/00034167d525f22089f34555531241beff3d75073618e7169c
Successfully built qtile
Installing collected packages: qtile
  Attempting uninstall: qtile
    Found existing installation: qtile 0.19.0
    Uninstalling qtile-0.19.0:
      Successfully uninstalled qtile-0.19.0
Successfully installed qtile-0.19.0
```

Everything looks good to me so far but I'm unable to actually get Qtile to start.
`xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.`

I've tried different commits, with and without my qtile config.

**Comment by jamestrew** [comment]
Here it is
```
d7363d9b763add64ef515484ebd09d370761c950 is the first bad commit
commit d7363d9b763add64ef515484ebd09d370761c950
Author: elParaguayo <elparaguayocode@gmail.com>
Date:   Sun Jan 16 09:17:44 2022 +0000

    Fix reconfiguration of bars on screen_change event

    Previously, once the bar was configured, any future configurations
    were blocked. This was a consequence of trying to stop margins being
    added multiple times.

    This PR addresses this by resetting the `configured` flag when a
    screen_change event arises.

    Fixes #3039
    Fixes #3148
    Fixes #3150

 CHANGELOG                |  1 +
 libqtile/bar.py          | 14 ++++++++++----
 libqtile/config.py       |  5 ++++-
 libqtile/core/manager.py |  4 ++--
 4 files changed, 17 insertions(+), 7 deletions(-)
```

**Comment by elParaguayo** [comment]
Cool. Thanks. Always nice to see my own name on the offending commit 😉 

Let me look at fixing this.

**Comment by jamestrew** [comment]
:grin: 
Thank you. Appreciate it.

**Comment by elParaguayo** [comment]
I'm pretty sure I know what I did wrong too...

**Comment by elParaguayo** [comment]
Right - there's a fix in #3278 if you're able to test it.

**Comment by jamestrew** [comment]
#3278 resolves this issue :+1:
